src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -341,6 +341,128 @@ public BitArray Not()
             return this;
         }
 
+        /*=========================================================================
+        ** Shift all the bit values to right on count bits. When count less zero it
+        ** shift all the bit values to left. New bits are be set to zero. The 
+        ** current instance is updated and returned.
+        =========================================================================*/
```

Comment:
@AlexRadch 

```c#
        int x = 8;
        Console.WriteLine(x << 1);
        Console.WriteLine(x << -1);
        Console.WriteLine(x << 0);
        Console.WriteLine(x >> 1);
        Console.WriteLine(x >> -1);
        Console.WriteLine(x >> 0);
```

gives output,

```
16
0
8
4
0
8
```

The c# shift operators just return 0 on negative count. But on BitArray, making it 0 would not be a desired behavior in manipulating the array in place to an unexpected value. Same case for changing the shift direction as well, when the API explicitly mentions right shift, and the array being manipulated in a different shift. I would prefer if an exception is thrown.